dashboard/src/main/java/org/openedx/courses/presentation/AllEnrolledCoursesViewModel.kt

```diff
@@ -0,0 +1,185 @@
+package org.openedx.courses.presentation
+
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.openedx.core.BaseViewModel
+import org.openedx.core.R
+import org.openedx.core.UIMessage
+import org.openedx.core.config.Config
+import org.openedx.core.domain.model.EnrolledCourse
+import org.openedx.core.extension.isInternetError
+import org.openedx.core.system.ResourceManager
+import org.openedx.core.system.connection.NetworkConnection
+import org.openedx.core.system.notifier.CourseDashboardUpdate
+import org.openedx.core.system.notifier.DiscoveryNotifier
+import org.openedx.dashboard.domain.CourseStatusFilter
+import org.openedx.dashboard.domain.interactor.DashboardInteractor
+import org.openedx.dashboard.presentation.DashboardAnalytics
+import org.openedx.dashboard.presentation.DashboardRouter
+
+class AllEnrolledCoursesViewModel(
+    private val config: Config,
+    private val networkConnection: NetworkConnection,
+    private val interactor: DashboardInteractor,
+    private val resourceManager: ResourceManager,
+    private val discoveryNotifier: DiscoveryNotifier,
+    private val analytics: DashboardAnalytics,
+    val dashboardRouter: DashboardRouter
+) : BaseViewModel() {
+
+    val apiHostUrl get() = config.getApiHostURL()
+    val hasInternetConnection: Boolean
+        get() = networkConnection.isOnline()
+
+    private val coursesList = mutableListOf<EnrolledCourse>()
+    private var page = 1
+    private var isLoading = false
+
+    private val _uiState = MutableStateFlow<AllEnrolledCoursesUIState>(AllEnrolledCoursesUIState.Loading)
+    val uiState: StateFlow<AllEnrolledCoursesUIState>
+        get() = _uiState.asStateFlow()
+
+    private val _uiMessage = MutableSharedFlow<UIMessage>()
+    val uiMessage: SharedFlow<UIMessage>
+        get() = _uiMessage.asSharedFlow()
+
+    private val _updating = MutableStateFlow<Boolean>(false)
```

Comment:
why _updating and _canLoadMore are in the separate StateFlow, maybe we can put them into UIState?

Reply:
We used this style in all view models. I think it's an important problem that we don't have one documented standard. Maybe we can discuss this in the future and refactor all view models.

Reply:
I don't agree that this style is acceptable. Decreasing a number of the flow subscriptions should increase a general speed of the application